File: alternate_capitals.rb
# Given a string, write a function that returns a copy of the original string that has every other character capitalized. (Capitalization should begin with the second character.)

#   Input: “hello, how are your porcupines today?”
#   Output: “hElLo, HoW ArE YoUr pOrCuPiNeS ToDaY?”

def alternate_capitals(string)
  output = string.downcase
  index = 1
  while index < output.length
    output[index] = output[index].upcase
    index += 2
  end
  return output
end

p alternate_capitals("hello, how are your porcupines today?")

File: c++/alternate_capitals.cpp
// Given a string, write a function that returns a copy of the original string that has every other character capitalized. (Capitalization should begin with the second character.)

// Input: “hello, how are your porcupines today?”
// Output: “hElLo, HoW ArE YoUr pOrCuPiNeS ToDaY?”

#include <iostream>
#include <string>
using namespace std;

string capitals(string input);

int main()
{
  cout << capitals("hello, how are your porcupines today?") << endl;
  return 0;
}

string capitals(string input)
{
  char current_letter;
  string output;
  for (int i = 0; i < input.length(); i++)
  {
    if (i % 2 == 0)
    {
      current_letter = tolower(input[i]);
    }
    else
    {
      current_letter = toupper(input[i]);
    }
    output = output + current_letter;
  }
  return output;
}